subsys/usb/device_next/class/usbd_msc_scsi.c

```diff
@@ -0,0 +1,832 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/storage/disk_access.h>
+#include <zephyr/sys/byteorder.h>
+
+#include "usbd_msc_scsi.h"
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_DECLARE(usbd_msc, CONFIG_USBD_MSC_LOG_LEVEL);
+
+#define INQUIRY_VERSION_SPC_2	0x04
+#define INQUIRY_VERSION_SPC_3	0x05
+#define INQUIRY_VERSION_SPC_4	0x06
+#define INQUIRY_VERSION_SPC_5	0x07
+
+/* Claim conformance to SPC-2 because this allows us to implement less commands
+ * and do not care about multiple reserved bits that became actual options
+ * later on. DO NOT change unless you make sure that all mandatory commands are
+ * implemented and all options (e.g. vpd pages) that are mandatory at given
+ * version are also implemented.
+ */
+#define CLAIMED_CONFORMANCE_VERSION INQUIRY_VERSION_SPC_2
+
+#define T10_VENDOR_LENGTH	8
+#define T10_PRODUCT_LENGTH	16
+#define T10_REVISION_LENGTH	4
+
+/* Optional, however Windows insists on reading Unit Serial Number.
+ * There doesn't seem to be requirement on minimum product serial number length,
+ * however when the number is not available the device shall return ASCII spaces
+ * in the field.
+ */
+#define UNIT_SERIAL_NUMBER "  "
+
+/* Every SCSI command has to abide to the general handling rules. Use macros
+ * to allow generating boilerplate handling code.
+ */
+#define SCSI_CMD_STRUCT(opcode) struct scsi_##opcode##_cmd
+#define SCSI_CMD_HANDLER(opcode)					\
+static int scsi_##opcode(struct scsi_ctx *ctx,				\
+			 struct scsi_##opcode##_cmd *cmd,		\
+			 uint8_t data_in_buf[CONFIG_USBD_MSC_SCSI_BUFFER_SIZE])
```

Comment:
Looks like there is not data_out_buf, I suggest to name just data, or buf. 
```suggestion
			 uint8_t *const data)
```
or one that generates a warning
```suggestion
			 uint8_t data[static CONFIG_USBD_MSC_SCSI_BUFFER_SIZE])
```

Reply:
> "Data-In Buffer" and "Data-Out Buffer" are the terms used in SAM-6 (5.4.2.2 Send SCSI Command SCSI transport protocol service request) so I refuse to change the name.

okay, but why not `uint8_t data_in_buf[static CONFIG_USBD_MSC_SCSI_BUFFER_SIZE])`, otherwise I do not see a reason to use sized array parameter.

Reply:
I had to look it up what `static` means in this context and it means what I really want. Added `static` to all declarations.